frontend/src/components/crd/CustomResourceInstancesList.tsx

```diff
@@ -10,8 +10,14 @@ import { ResourceListView } from '../common/Resource';
 function CrInstancesView({ crds }: { crds: CRD[]; key: string }) {
   const { t } = useTranslation(['glossary', 'translation']);
 
-  const crdInfo = useMemo(() => crds.map(crd => crd.makeCRClass()), [crds]);
-  const queries = crdInfo.map(cr => cr.useList());
+  const dataClassCrds = crds.map(crd => {
+    const crdClass = crd.makeCRClass();
+    const data = crdClass.useList({ cluster: crd.cluster });
+    return { data, crdClass, crd };
+  });
```

Comment:
Should we keep a useMemo for at least generating the classes here?

Reply:
I don't think so. I had tried that, but generating the classes doesn't take very much time.

If it's a problem for you, and not a nit... I can have a bit more of a think.

Reply:
I've expanded on the commit message.

Reply:
> I don't think so. I had tried that, but generating the classes doesn't take very much time.
> 
> If it's a problem for you, and not a nit... I can have a bit more of a think.

Alright. It was mostly a nit. We can keep it like this and optimize later if needed.